### PR TITLE
Annotate more functions with explicit nulls

### PR DESCRIPTION
**Pull request by lexidor**

### CODE REVIEW

**Comment by lexidor** [comment]
Please no merge to master anymore ;).

File: src/classes/Trades.ts [hunk 1]
```diff
@@ -570,6 +570,8 @@ export = class Trades {
                                     return reject(err);
                                 }
 
+                                // @nicklason you know that match might be undefined here?
+                                // It doesn't look like your code can handle undefined.
```
Comment (Nicklason) [inline src/classes/Trades.ts]: it is fixed in development

**Comment by lexidor** [comment]
> Please no merge to master anymore ;).

Ouch, the pain!
This PR just became 5 times its size...

File: src/classes/Trades.ts (hunk 1 above)
Comment (lexidor) [reply src/classes/Trades.ts]: Many nice

**Comment by Nicklason** [comment]
> > Please no merge to master anymore ;).
> 
> Ouch, the pain!
> This PR just became 5 times its size...

The bigger the better

**Comment by lexidor** [comment]
> > > Please no merge to master anymore ;).
> > 
> > 
> > Ouch, the pain!
> > This PR just became 5 times its size...
> 
> The bigger the better

I want to make eeny beeny tiny commities. UwU

Just kiddin', but large PRs are a pain to review.

**Comment by lexidor** [comment]
> Ohh nooooes! Repo can't find any persons named "Nicklason". Please, choose another type of person.

:rofl:

File: src/classes/Commands.ts [hunk 2]
```diff
@@ -916,15 +917,15 @@ export = class Commands {
...
+        if (typeof params.buy === 'object' && params.buy !== null) {
             params.buy.keys = params.buy.keys || 0;
             params.buy.metal = params.buy.metal || 0;
 
             if (params.autoprice === undefined) {
                 params.autoprice = false;
             }
         }
-        if (typeof params.sell === 'object') {
+        if (typeof params.sell === 'object' && params.sell !== null) {
```
Comment (Nicklason) [inline src/classes/Commands.ts]: I don't really like this, the bot does not support null prices.

I'll work on a fix and make a pull request.

**Pull request merged**